### DISCUSSION TITLE
v0.13.0

### DISCUSSION THREAD

**Discussion by romange** [dragonflydb/dragonfly#629] [Announcements]
# Dragonfly v0.13.0

The first release of the year! Yey! 👯

This release contains a few exciting features.
- Active defrag: any long-running process with non-trivial allocation patterns suffers from excessive memory usage due to external fragmentation. This release uses an improved defragmentation algorithm after we released the initial version in the previous version.
- Experimental SSD-based data-tiering support. Dragonfly is the only in-memory store that provides data tiering that is faster than other memory-only stores. (Currently only for strings). Try it with flag `--spill_file_prefix=<nvme_file_basepath>`
- JSON - this release compliments the already existing JSON API with support for SCAN and TYPE working with JSON data types.  
- Replication (still not for prod use) but can already be tried on simple single key commands like SET, SADD, ZADD, LPUSH etc.  
- Writes and reads of large string values are optimized for x86 architectures with SIMD instructions.
- Numerous improvements to helm templates, plus docker health checks - thanks to @tamcore !

## What's Changed
* feat(rdb save): Create compressor interface by @adiholden in https://github.com/dragonflydb/dragonfly/pull/538
* fix(helm): Chart.yaml - Add home, keywords, sources, kubeVersion by @arukiidou in https://github.com/dragonflydb/dragonfly/pull/542
* feat: add healtcheck for container by @tamcore in https://github.com/dragonflydb/dragonfly/pull/544
* Support lz4 compression by @adiholden in https://github.com/dragonflydb/dragonfly/pull/545
* feat(server): json set type support by @boazsade in https://github.com/dragonflydb/dragonfly/pull/546
* chore: Update helio dependency by @romange in https://github.com/dragonflydb/dragonfly/pull/553
* chore: Update helio dependency by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/555
* feat(server): Implement robust error & cancellation on replica by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/531
* feat(ci test): add build and test on release mode by @adiholden in https://github.com/dragonflydb/dragonfly/pull/556
* added example of start script by @Tomato6966 in https://github.com/dragonflydb/dragonfly/pull/557
* Support offloading of blobs longer than 2KB by @romange in https://github.com/dragonflydb/dragonfly/pull/559
* feat(server): active memory defrag test application by @boazsade in https://github.com/dragonflydb/dragonfly/pull/552
* feat(server): JSON family using JSON type by @boazsade in https://github.com/dragonflydb/dragonfly/pull/561
* feat(server): New auto-journal types/read/write by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/560
* chore: Refactor RdbLoad by @romange in https://github.com/dragonflydb/dragonfly/pull/564
* chore: Adding a mpsc intrusive queue based on Vyukov's design by @romange in https://github.com/dragonflydb/dragonfly/pull/562
* feat: introduce simd algorithm for bitpacking by @romange in https://github.com/dragonflydb/dragonfly/pull/568
* feat: implement ascii_unpack using SIMD instructions by @romange in https://github.com/dragonflydb/dragonfly/pull/573
* bug(rdb load): Rdb loader tasks stop running on failure #567 by @adiholden in https://github.com/dragonflydb/dragonfly/pull/576
* chore: Improve the implementation of simd based packing by @romange in https://github.com/dragonflydb/dragonfly/pull/577
* fix: choose an appropriate SIMD implementation for aarch64 by @romange in https://github.com/dragonflydb/dragonfly/pull/579
* fix: handle corner case when comitted memory is negative by @romange in https://github.com/dragonflydb/dragonfly/pull/570
* feat(server): Use new journal format by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/563
* fix(chart): probe indentation by @tamcore in https://github.com/dragonflydb/dragonfly/pull/590
* feat(server): Redesign and simplify tiered storage module. by @romange in https://github.com/dragonflydb/dragonfly/pull/589
* use dense_set for hashmap family by @romange in https://github.com/dragonflydb/dragonfly/pull/593
* feat(server): adding version monitoring task by @boazsade in https://github.com/dragonflydb/dragonfly/pull/586
* feat(replica): Support FlushDB command for replication #580 by @adiholden in https://github.com/dragonflydb/dragonfly/pull/591
* fix(server): defrag unit test fix by @boazsade in https://github.com/dragonflydb/dragonfly/pull/602
* chore(server): upgrade mimalloc for v2.0.7 by @boazsade in https://github.com/dragonflydb/dragonfly/pull/597
* feat: build and publish weekly alpha/development container image by @tamcore in https://github.com/dragonflydb/dragonfly/pull/594
* life of a transaction by @romange in https://github.com/dragonflydb/dragonfly/pull/278
* feat(server): track hits and misses stats. by @ashotland in https://github.com/dragonflydb/dragonfly/pull/608
* feat(server): Improved cancellation by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/599
* feat(replica): atomicity for multi shard commands by @adiholden in https://github.com/dragonflydb/dragonfly/pull/598
* fix(server):set of existing object with expiry by @ashotland in https://github.com/dragonflydb/dragonfly/pull/609
* chore: Add tiered_storage_test. by @romange in https://github.com/dragonflydb/dragonfly/pull/613
* feat(tools):Cache logs player by @ashotland in https://github.com/dragonflydb/dragonfly/pull/614
* fix(tools): cache logs player handle append as redis append by @ashotland in https://github.com/dragonflydb/dragonfly/pull/615
* docs(quickstart): Clarify docker command for macOS by @pjambet in https://github.com/dragonflydb/dragonfly/pull/612
* feat(tools):cache log player batching all the way optimisation by @ashotland in https://github.com/dragonflydb/dragonfly/pull/617
* feat(ci test): add testing for helm chart by @tamcore in https://github.com/dragonflydb/dragonfly/pull/622
* chore: overhaul chart by @tamcore in https://github.com/dragonflydb/dragonfly/pull/620
* Cache logs player - add one last stats print after completion by @ashotland in https://github.com/dragonflydb/dragonfly/pull/623
* feat(server): Buffered journal serializers by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/619
* feat(server): Update helio, optimize and clean up rdb/snapshot by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/625
* fix(server): only print version if successfully connect. by @boazsade in https://github.com/dragonflydb/dragonfly/pull/628
* feat(server): using memory defrag with per shard mem info by @boazsade in https://github.com/dragonflydb/dragonfly/pull/616

## New Contributors
* @arukiidou made their first contribution in https://github.com/dragonflydb/dragonfly/pull/542
* @ashotland made their first contribution in https://github.com/dragonflydb/dragonfly/pull/608
* @pjambet made their first contribution in https://github.com/dragonflydb/dragonfly/pull/612

**Full Changelog**: https://github.com/dragonflydb/dragonfly/compare/v0.12.0...v0.13.0

<hr /><em>This discussion was created from the release <a href='https://github.com/dragonflydb/dragonfly/releases/tag/v0.13.0'>v0.13.0</a>.</em>

**Comment by snird** [comment]
SSD data-tiering is a killer feature!
Awesome release @romange and the team.
